src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,124 @@
 package core.basesyntax;
 
+import java.util.Map;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int GROW_FACTOR = 2;
+    private static final double LOAD_FACTOR = 0.75d;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = createStartTable();
+    }
 
     @Override
     public void put(K key, V value) {
-
+        Node<K, V> existingNode = getNode(key);
+        if (existingNode != null) {
+            existingNode.value = value;
+            return;
+        }
+        double currentFactor = ((double) size / table.length);
+        if (currentFactor >= LOAD_FACTOR) {
+            resize();
+        }
+        int index = calculateBucketPosition(key);
+        Node<K, V> entryNode = new Node<>(key, value, null);
+        if (table[calculateBucketPosition(key)] != null) {
+            entryNode.next = table[index];
+        }
+        table[index] = entryNode;
+        size++;
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> targetNode = getNode(key);
+        return targetNode == null ? null : targetNode.getValue();
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> implements Map.Entry<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+
+        @Override
+        public K getKey() {
+            return key;
+        }
+
+        @Override
+        public V getValue() {
+            return value;
+        }
+
+        @Override
+        public V setValue(V newValue) {
+            V oldValue = value;
+            value = newValue;
+            return oldValue;
+        }
+    }
```

Comment:
[Don't use getters and setters in class Node](https://github.com/mate-academy/jv-hash-map/blob/master/checklist.md#dont-use-getters-and-setters-in-class-node)